spec/support/database.rb

```diff
@@ -1,13 +1,45 @@
 # frozen_string_literal: true
 
-module TemporalTables
-  module DatabaseAdapter
-    def self.adapter_name
-      if Gemika::Env.gem?('pg')
-        'postgresql'
-      elsif Gemika::Env.gem?('mysql2')
-        'mysql'
-      end
+def adapter_name
+  if Gemika::Env.gem?('pg')
+    'postgresql'
+  elsif Gemika::Env.gem?('mysql2')
+    'mysql'
+  else
+    raise 'Cannot determine adapter'
+  end
+end
+
+def table_exists?(name)
+  ActiveRecord::Base.connection.table_exists?(name)
+end
+
+def function_exists?(name)
+  case adapter_name
+  when 'postgresql'
+    begin
+      ActiveRecord::Base.connection.execute("select(pg_get_functiondef('#{name}'::regprocedure))")
+      true
+    rescue ActiveRecord::StatementInvalid
+      false
     end
+  when 'mysql'
+    raise NotImplementedError, 'TODO: Implement for mysql'
+  else
+    raise "Unknown adapter #{adapter_name}"
   end
 end
+
+def trigger_exists?(name)
+  case adapter_name
+  when 'postgresql'
+    ActiveRecord::Base.connection.execute(
+      "select (pg_get_triggerdef(oid)) FROM pg_trigger WHERE tgname = '#{name}'"
+    ).first.present?
+  when 'mysql'
+    raise NotImplementedError, 'TODO: Implement for mysql'
```

Comment:
... same for trigger: Can somebody with more mysql experience maybe provide a way how to check if a trigger is defined?

Reply:
I got this one working for mysql